crates/astria-sequencer/src/app.rs

```diff
@@ -387,9 +399,28 @@ impl App {
                 continue;
             }
 
+            // Don't include tx if it would make the cometbft block too large
+            if let Some(remaining) = remaining_cometbft_bytes {
+                if remaining < i64::try_from(tx.len()).expect("failed to turn tx length into i64") {
+                    debug!(
+                        transaction_hash = %telemetry::display::hex(&tx_hash),
+                        remaining_cometbft_bytes = remaining,
+                        tx_len = tx.len(),
+                        "excluding transaction: max cometbft data limit reached"
+                    );
+                    excluded_tx_count += 1;
+                    continue;
```

Comment:
at this point we know we can't include any more txs, so just update `excluded_tx_count += remaining_tx_count` and return - you can update `for tx in txs` to `for (i, tx) in txs.enumerate()` to get the tx index you're at